### PR TITLE
Update to support latest docker API

### PR DESCRIPTION
**Pull request by flenter**
* updates fsouza/go-dockerclient
* move HostConfig to CreateContainer calls (as also described in https://github.com/wercker/wercker/issues/187#issuecomment-227372248)
* switched from `CopyFromContainer` to `DownloadFromContainer`

### CODE REVIEW

File: vendor/vendor.json [hunk 1]
```diff
@@ -556,10 +580,10 @@
 			"revisionTime": "2016-04-12T13:37:56Z"
 		},
 		{
-			"checksumSHA1": "SzhvLPReGgCOq1FhoK474deJIPc=",
+			"checksumSHA1": "hpj0xzuu165n/5qY5T0YDRxrzuQ=",
 			"path": "github.com/fsouza/go-dockerclient",
-			"revision": "681783e5c84bb9d669f52814abd42297e058e25b",
-			"revisionTime": "2016-04-25T15:38:51Z"
+			"revision": "107dc3807d24e581e874212dd9a7ba35599b893d",
```
Comment (hatchan) [inline vendor/vendor.json]: All other `github.com/fsouza/go-dockerclient/*` depdendencies still use `681783e5c84bb9d669f52814abd42297e058e25b`. Shouldn't these also be updated to `107dc3807d24e581e874212dd9a7ba35599b893d`?

File: vendor/vendor.json [hunk 2]
```diff
@@ -388,6 +394,12 @@
 			"revisionTime": "2016-04-25T21:02:26Z"
 		},
 		{
+			"checksumSHA1": "iQlaOTsq2en8kZrOt7pWWVGO+kk=",
+			"path": "github.com/docker/docker/pkg/homedir",
```
Comment (hatchan) [inline vendor/vendor.json]: Not all `github.com/docker/docker` have been updated, shouldn't we upgrade all of them?

**Comment by hatchan** [comment]
Changes look fine, just a bit concerned about using different versions of the same repository in the `vendor.json`

**Comment by hatchan** [comment]
Also don't forget to update changelog.md

**Comment by flenter** [comment]
There were several dependencies that are no longer used (according to govendor). I used `govendor remove +u` to clean up the `vendor.json` file. 

I've also updated github.com/docker/docker, github.com/docker/distribution and github.com/docker/engine-api dependencies in the vendor.json (all repos under the docker user except go-connections and go-units).

**Comment by flenter** [comment]
If I run `test-all.sh` it fails, I'm not sure if that was already the case since I only have the latest version of docker running on my machine. It fails with: 

```
testing source-path... passed
testing local services... passed
testing deploy... passed
testing deploy target... passed
testing after steps... passed
testing relative symlinks... passed
testing after steps fail... passed
testing empty wercker file... passed
testing multiple services with the same image... failed
time="2016-06-27T13:43:01+02:00" level=info line="options.go:48" msg="--> No Docker host specified, checking: /var/run/docker.sock" Logger=docker
```
And finally:
```
time="2016-06-27T13:43:22+02:00" level=debug line="events.go:177" msg=BuildStepFinished Logger=Events
time="2016-06-27T13:43:22+02:00" level=debug line="events.go:169" msg="  ArtifactURL string = " Logger=Events
time="2016-06-27T13:43:22+02:00" level=debug line="events.go:164" msg="  Box core.Box" Logger=Events
time="2016-06-27T13:43:22+02:00" level=debug line="events.go:169" msg="  Build core.Pipeline = <nil>" Logger=Events
time="2016-06-27T13:43:22+02:00" level=debug line="events.go:169" msg="  Message string = no such image" Logger=Events
time="2016-06-27T13:43:22+02:00" level=debug line="events.go:169" msg="  Order int = 2" Logger=Events
time="2016-06-27T13:43:22+02:00" level=debug line="events.go:169" msg="  PackageURL string = " Logger=Events
time="2016-06-27T13:43:22+02:00" level=debug line="events.go:164" msg="  Step core.Step" Logger=Events
time="2016-06-27T13:43:22+02:00" level=debug line="events.go:169" msg="    BaseStep *core.BaseStep = &{ <nil>  setup environment wercker setup environment 1.0.0  }" Logger=Events
time="2016-06-27T13:43:22+02:00" level=debug line="events.go:160" msg="    data map[string]string = <not exported>" Logger=Events
time="2016-06-27T13:43:22+02:00" level=debug line="events.go:160" msg="    logger *util.LogEntry = <not exported>" Logger=Events
time="2016-06-27T13:43:22+02:00" level=debug line="events.go:160" msg="    options *core.PipelineOptions = <not exported>" Logger=Events
time="2016-06-27T13:43:22+02:00" level=debug line="events.go:160" msg="    stepDesc *core.StepDesc = <not exported>" Logger=Events
time="2016-06-27T13:43:22+02:00" level=debug line="events.go:160" msg="    url string = <not exported>" Logger=Events
time="2016-06-27T13:43:22+02:00" level=debug line="events.go:169" msg="  Successful bool = false" Logger=Events
```

**Comment by termie** [comment]
i feel like this won't work with the version of docker we run on the servers, do we have any evidence that it does?

**Comment by flenter** [comment]
Just an update: We've been running this version on our staging servers for a day or so now and haven't seen any issues with it. We're currently investigating whether the `test-all.sh` script/test was already failing before this change.

**Comment by Tonnu** [comment]
Just ran the `test-all.sh` script using docker v1.10 and it's failing on `testing multiple services with the same image`, but with a different error:

```
time="2016-07-01T10:14:52+02:00" level=info line="asm_amd64.s:472" msg="[ ]  stdin \"echo 92bf3838-800c-4ee9-81d3-91e5cebcf90e $?\\n\"" Hidden=true Logger=Literal Stream=stdin
time="2016-07-01T10:14:52+02:00" level=info line="asm_amd64.s:472" msg="[x] stdout \"go vet for .\\n\"" Hidden=false Logger=Literal Stream=stdout
time="2016-07-01T10:14:52+02:00" level=info line="asm_amd64.s:472" msg="[x] stdout \"go vet for ./.wercker\\n\"" Hidden=false Logger=Literal Stream=stdout
time="2016-07-01T10:14:52+02:00" level=info line="asm_amd64.s:472" msg="[x] stdout \"Usage of /usr/local/go/pkg/tool/linux_amd64/vet:\\n\\tvet [flags] directory...\\n\\tvet [flags]
```
and later on:
```
time="2016-07-01T10:14:59+02:00" level=warning line="box.go:509" msg="Box container has already stopped." Logger=Box Name="golang:latest" ShortName=golang
time="2016-07-01T10:14:59+02:00" level=debug line="box.go:454" msg="Removing container: 90796494f746102b8131f45d1ba9b30d4ca9fb2cceccbc3c9408cee1b6d0c741" Container="90796494f746102b8131f45d1ba9b30d4ca9fb2cceccbc3c9408cee1b6d0c741" Logger=Box Name="golang:latest" ShortName=golang
time="2016-07-01T10:15:00+02:00" level=debug line="events.go:177" msg=FullPipelineFinished Logger=Events
time="2016-07-01T10:15:00+02:00" level=debug line="events.go:169" msg="  AfterStepSuccessful bool = false" Logger=Events
time="2016-07-01T10:15:00+02:00" level=debug line="events.go:169" msg="  MainSuccessful bool = false" Logger=Events
time="2016-07-01T10:15:00+02:00" level=debug line="events.go:169" msg="  RanAfterSteps bool = false" Logger=Events
time="2016-07-01T10:15:00+02:00" level=fatal line="main.go:71" msg="Step failed: go vet" Logger=CLI
```

So looks like `go vet` is being executed with the wrong params. I guess that does mean that the actual starting of multiple services with the same image name actually used to work on `v1.10`.

**Comment by Tonnu** [comment]
So it looks like testing multiple images with the same name never actually worked (see #202). Think this PR is good to go?

**Pull request merged**